components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt

```diff
@@ -16,17 +18,50 @@ import mozilla.components.support.base.observer.ObserverRegistry
 @Suppress("TooManyFunctions")
 class SessionManager(
     val engine: Engine,
+    val initialSessions: SessionsSnapshot? = null,
+    val defaultSession: (() -> Session)? = null,
     delegate: Observable<SessionManager.Observer> = ObserverRegistry()
 ) : Observable<SessionManager.Observer> by delegate {
     private val values = mutableListOf<Session>()
+    // Access must be guarded by synchronizing on `values`.
     private var selectedIndex: Int = NO_SELECTION
 
+    init {
+        initialSessions?.forEach {
+            add(it.session, selected = it.selected, engineSession = it.engineSession)
+        }
+
+        if (defaultSession != null && size == 0) {
```

Comment:
I wouldn't apply this on init, because there the caller specifies directly what the initial sessions are. Either based on what is restored from storage or based on manually setting it (see my comment on the sample app.)